patches/chromium/feat_add_set_theme_source_to_allow_apps_to.patch

```diff
@@ -66,26 +62,28 @@ index 70389e0245993faa2c17e9deefeb000280ef2368..4d56998a391c1c7508bb911eea0c16cd
  };
  
 diff --git a/ui/native_theme/native_theme_dark_aura.cc b/ui/native_theme/native_theme_dark_aura.cc
-index a8fbfee3b13672902aac05fd5a65fa8ee81f9f7e..58ee87151f40229776ea356eaa6adc30eca389be 100644
+index a8fbfee3b13672902aac05fd5a65fa8ee81f9f7e..1be6369acf0b7c02a6f862636c2b2de1fbf8cb5a 100644
 --- a/ui/native_theme/native_theme_dark_aura.cc
 +++ b/ui/native_theme/native_theme_dark_aura.cc
-@@ -20,6 +20,7 @@ SkColor NativeThemeDarkAura::GetSystemColor(ColorId color_id,
+@@ -20,6 +20,8 @@ SkColor NativeThemeDarkAura::GetSystemColor(ColorId color_id,
  }
  
  bool NativeThemeDarkAura::ShouldUseDarkColors() const {
-+  RETURN_IF_FORCED_THEME
++  if (theme_source() == ThemeSource::kForcedLight) return false;
++  if (theme_source() == ThemeSource::kForcedDark) return true;
    return true;
  }
  
 diff --git a/ui/native_theme/native_theme_win.cc b/ui/native_theme/native_theme_win.cc
-index 3003643bfb78cec2f5e84fc9e1471e1ef54aae41..f305ad3c809129a4b583e3bba04a8734931849b0 100644
+index 3003643bfb78cec2f5e84fc9e1471e1ef54aae41..06f2cbc84401958d49445f4ce6acb1b2fef0aa04 100644
 --- a/ui/native_theme/native_theme_win.cc
 +++ b/ui/native_theme/native_theme_win.cc
-@@ -611,6 +611,7 @@ bool NativeThemeWin::ShouldUseDarkColors() const {
+@@ -611,6 +611,8 @@ bool NativeThemeWin::ShouldUseDarkColors() const {
    // ...unless --force-dark-mode was specified in which case caveat emptor.
    if (UsesHighContrastColors() && !IsForcedDarkMode())
      return false;
-+  RETURN_IF_FORCED_THEME
++  if (theme_source() == ThemeSource::kForcedLight) return false;
++  if (theme_source() == ThemeSource::kForcedDark) return true;
```

Comment:
Isn't this handled by `NativeTheme::ShouldUseDarkColors`? Do we need to insert the logic twice?

Reply:
Yes but if this block changes to be more similar to the dark aura one (not call the base `NativeTheme`) I don't want to have that bug.  Safer to call in all places IMO and it doesn't cause any significant overhead. Future proofing against future chromium patches, would rather get a conflict than it mysteriously stop working.

Reply:
hm... won't our tests catch that if it happens?

Reply:
Depends on the conditions required.  i.e. if it breaks on windows when the native dark theme is enabled our tests can't catch that.